### ISSUE TITLE
Key-events "bleed through" to Kindle user interface on K3

### ISSUE THREAD

**Issue by ARF1** [hwhw/kindlevncviewer#9]
I am using this on K3 and have a problem, that did not occur with the non-jit vnc viewer:

When I have a running vnc connection on the K3 and I press a key, the key event is captured and sent to the remote machine as required.

BUT the key event is ALSO sent to the Kindle user interface (covered by the vnc viewer).

Say I press the arrow down key, the I see the line which marks the currently selected entry on the kindle start page "bleed through" the vnc image and move down the screen.

If I press enter, a ebook gets started (which then blocks out the vnc image until the vnc screen update), etc.

Is there any way to prevent the key event from being sent to the kindle UI? With the native version, this problem of the kindle UI reacting to keys pressed in the vnc viewer did not occur.

**Comment by hwhw** [comment]
Sorry, I lost track (and a bit of interest) of this project.
I have no explanation why the old version worked and the new doesn't. Maybe there's *other* changes in the system, too? (Like a newer firmware, different method to launch etc.pp.?) I'm not really in the situation to do my own tests, though - and even if I were I would be lacking enough time to get things done on the code, sorry.

**Comment by KlausBuchegger** [comment]
Hi, this is a relatively old issue, and I'm not sure if anyone is still interested, but I figured out why the key-event bleeding happened. 
I did some stracing to compare the ioctl calls of the old C approach and the new lua approach and I noticed that the ioctl call to grab the input devices failed failed.
This caused the key events to be shared by the vncviewer and the kindle UI.

I opened a pull request with a fix that works on my Kindle DXG.
